pkg/apis/agones/v1/gameserver.go

```diff
@@ -328,6 +333,10 @@ func (gss *GameServerSpec) applyPortDefaults() {
 		if p.Protocol == "" {
 			gss.Ports[i].Protocol = "UDP"
 		}
+
+		if p.Container == nil || *p.Container == "" {
```

Comment:
This should probably be wrapped in FeatureContainerPortAllocation, such that this value is not set when the flag is enabled. Since if people aren't using the alpha feature, it probably shouldn't show up as a value in the CRD.